workshops/login_auth/README.md

```diff
@@ -0,0 +1,538 @@
+
+Add the two lines of code below in the `if` condition of the code above.
+
+```javascript
+document.getElementById("user_div").style.display = "block";
+document.getElementById("login_div").style.display = "none";
+```
+
+The code should look like this:
+
+```javascript
+firebase.auth().onAuthStateChanged(function(user) {
+  if (user) {
+    // User is signed in.
+    document.getElementById("user_div").style.display = "block";
+    document.getElementById("login_div").style.display = "none";
+  } else {
+    // No user is signed in.
+  }
+});
+```
+
+In the code above `block` will display the division and `none` will hide it.
+
+You added the part for the signed-in user so you now need the `else` part. This is where the user isn't signed in. The code will be the same as the code in the first `if` but you need to swap `block` and `none` to display only `main_div` with ID `login_div`.
+
+Add the following code to the `else` condition:
+
+```javascript
+document.getElementById("user_div").style.display = "none";
+document.getElementById("login_div").style.display = "block";
+```
+
+The javaScript code should look like:
+
+```javascript
+firebase.auth().onAuthStateChanged(function(user) {
+  if (user) {
+    // User is signed in.
+    document.getElementById("user_div").style.display = "block";
+    document.getElementById("login_div").style.display = "none";
+  } else {
+    // No user is signed in.
+    document.getElementById("user_div").style.display = "none";
+    document.getElementById("login_div").style.display = "block";
+  }
+});
+```
+Now, how it looks like:
+
+![with JS](https://cloud-e3wysrojq.vercel.app/c1.png)
+
+Now you need to work on both the functions you assigned to HTML buttons (`login()` and `logout()`).
+
+##### Function login()
+A user can sign-in only if their record is stored in Firebase. So, firebase has a property called [`Sign-in existing user`](https://firebase.google.com/docs/auth/web/start#sign_in_existing_users) as existing users can sign-in through input fields.
+
+Now you need to create two variables in `login()` named `userEmail` and `userPass` which get email and password values from the input fields with IDs `email_field` and `password_field`.
+
+```javascript
+function login(){
+
+  var userEmail = document.getElementById("email_field").value;
+  var userPass = document.getElementById("password_field").value;
+}
+```
+So in the function above, you need to add the [`Sign-in existing user`](https://firebase.google.com/docs/auth/web/start#sign_in_existing_users) property.
+
+```javascript
+firebase.auth().signInWithEmailAndPassword(email, password).catch(function(error) {
+  // Handle Errors here.
+  var errorCode = error.code;
+  var errorMessage = error.message;
+  // ...
+});
+```
+In the code above replace `email` with `userEmail` and `password` with `userPass`.
+
+You also need to display an error message when an error occurs like _wrong email or password_, so add the below code in the above function.
```

Comment:
```suggestion
You also need to display an error message when an error occurs like _wrong email or password_, so add the code below in the above function.
```

Reply:
Detail if the reader should be adding it to the function call of `signInWithEmailAndPassword` or the body of the `login` function.